app/views/case_contacts/index.html.erb

```diff
@@ -15,35 +15,47 @@
   </div>
 </div>
 
-<%= form_for_filterrific @filterrific, url: case_contacts_path do |f| %>
+<%= form_for_filterrific @filterrific, url: case_contacts_path, html: {class: "my-4"} do |f| %>
   <%= hidden_field_tag 'casa_case_id', params[:casa_case_id] %>
-  <div class="card-style my-4">
-    <div class="card-content d-flex justify-content-between align-items-end">
-      <div class="h4 btn-lg">Filter by</div>
-      <button href="#"
-              class="btn btn-lg btn-link"
-              type="button"
-              data-bs-toggle="collapse"
-              data-bs-target="#filter-card-body"
-              aria-expanded="<%= params[:filterrific].present? %>>"
-              aria-controls="filter-card-body">
-        <div class="h4">Show / Hide</div>
-      </button>
-    </div>
 
-  <% collapse_class = params[:filterrific] ? "" : "collapse" %>
-    <div class="card-content mb-10 ml-20 <%= collapse_class %>" id="filter-card-body">
-      <div class="row">
-        <div class="col-12">
-          <h3 class="mt-10"><label>Draft status</label></h3>
-        </div>
-        <div class="col-md-4">
+  <div class="card-style">
+    <div class="card-content">
+      <div class="d-flex justify-content-between align-items-center">
+        <div class="h4 btn-lg">Filter by</div>
+        <button href="#"
+                class="btn btn-lg btn-link"
+                type="button"
+                data-bs-toggle="collapse"
+                data-bs-target="#filter-card-body"
+                aria-expanded="<%= params[:filterrific].present? %>>"
+                aria-controls="filter-card-body">
+          <div class="h4">Expand / Hide</div>
+        </button>
+      </div>
+      <div class="d-flex flex-wrap justify-content-between align-items-end gap-3 mb-4">
+        <div class="d-flex flex-column gap-3">
+          <div class="select-style-1 mb-0">
+            <%= f.label :sorted_by %>
+            <div class="select-position">
+              <%= f.select(:sorted_by, @filterrific.select_options[:sorted_by], {}, {class: ""}) %>
+            </div>
+          </div>
           <div class="form-check checkbox-style">
             <%= f.check_box :no_drafts, class: "form-check-input case-contact-contact-type" %>
             <label class="form-check-label" for="filterrific_no_drafts">Hide drafts</label>
           </div>
         </div>
+        <div>
+          <%= button_tag( :class=> "btn-sm main-btn dark-btn") do %>
+            <i class="lni lni-funnel mr-10"></i> Filter
+          <% end %>
+          <%= link_to("Reset filters", reset_filterrific_url, class: "btn-sm main-btn dark-btn-outline btn-hove") %>
+        </div>
       </div>
+    </div>
+
+    <% collapse_class = params[:filterrific] ? "collapse show" : "collapse" %>
+    <div class="card-content mb-10 ml-20 <%= collapse_class %>" id="filter-card-body">
```

Comment:
This creates a small issue where as soon as you use a filter the whole menu pops open and we want to avoid that for the commonly used filters.

![filter_popping_open](https://github.com/rubyforgood/casa/assets/14540596/033174ca-45c1-4db7-81aa-8c23302a6890)

I'll leave it up to you to decide how to fix it :slightly_smiling_face:

Reply:
@elasticspoon To clarify, do you want the filter collapse state to be agnostic from all filters or just the commonly used filters? I am thinking the former is nicer.

Currently on main, if you nav to the page with filter params, the menu will be expanded. Should the menu always be collapsed on loading the page?

Reply:
> @elasticspoon To clarify, do you want the filter collapse state to be agnostic from all filters or just the commonly used filters? I am thinking the former is nicer.

I think agnostic from all. 

> Currently on main, if you nav to the page with filter params, the menu will be expanded. Should the menu always be collapsed on loading the page?

So I think ideally: 
- I navigate to page => filter is closed
- I select a common filter, hit filter button => page updates filter menu remains closed
- I open the menu, select a some other filter, hit filter button => page updates, filter menu remains open

You _might_ have to add a turbo frame for this? Cause I am not sure how you could preserve filter openness state if a full page refresh is happening. 

If that is not possible I would rather it default to always closed.

Reply:
@elasticspoon I could see a turbo frame being beneficial for this case, but it looks like casa is not using turbo-rails. I could look at a solution that adds it, but maybe that is outside the scope of this PR?

I thought of a different approach: 28c8b1859b732650356527c9253fdbf33e8080c6. I made a Stimulus controller that encapsulates some of the BS collapse behavior, and it stores the state of the collapse menu in sessionStorage. 

> * I navigate to page => filter is closed
> * I select a common filter, hit filter button => page updates filter menu remains closed
> * I open the menu, select a some other filter, hit filter button => page updates, filter menu remains open

This fulfills the 2nd and 3rd cases you mentioned above, but for the 1st, instead the filter is open/closed depending on what the user left it as in the session.

I would be happy to look at adding turbo frames instead. In that approach, I don't think we would have to worry about storing the collapse state, since we would no longer be re-rendering the whole page.

Reply:
> I could see a turbo frame being beneficial for this case, but it looks like casa is not using turbo-rails.

Well then.... I am a bit embarrassed to learn that just now :blush: 

> I made a Stimulus controller that encapsulates some of the BS collapse behavior, and it stores the state of the collapse menu in sessionStorage.

That sounds pretty reasonable. I'll take a closer look soon. I would like to get this update in sooner rather than later. So while I do think turbo would provide the more ideal solution but I would much rather have the fix sooner. 

It would be dope if after this PR you could add in turbo-rails, but it is certainly out of the scope of **this** PR.

Reply:
> > I could see a turbo frame being beneficial for this case, but it looks like casa is not using turbo-rails.
> 
> Well then.... I am a bit embarrassed to learn that just now 😊
> 
> > I made a Stimulus controller that encapsulates some of the BS collapse behavior, and it stores the state of the collapse menu in sessionStorage.
> 
> That sounds pretty reasonable. I'll take a closer look soon. I would like to get this update in sooner rather than later. So while I do think turbo would provide the more ideal solution but I would much rather have the fix sooner.
> 
> It would be dope if after this PR you could add in turbo-rails, but it is certainly out of the scope of **this** PR.
😊 def understandable! There is a lot going on with this project, and I'm sure some parts are fairly old.

Reply:
I'm not sure I like session storage as the solution here. It overcomplicates this. The only reason is so that we can preserve the filters being opened after a user clicks the filter button, right? 

I agree that a turbo-frame is a better long-term fix and might be out of scope for this issue.

Reply:
@schoork thank you for the comments! I agree I am not happy with the session storage solution, but I wasn't sure of a cleaner way to persist client state between form submissions without it.

> The only reason is so that we can preserve the filters being opened after a user clicks the filter button, right?

That is correct.

I am unclear on the direction to take with all your comments together. To clarify, it seems to me that the two solutions you propose are either:

1. Revert to my previous solution. Not persist the collapse menu state and remove the Stimulus controller and just use Bootstrap for collapse behavior. Only downside really is that every time the user applies filters, the menu will collapse.
2. Refactor the Stimulus controller to set a key for the session storage. If Turbo is added, this controller will probably no longer be needed and deleted.

Is that correct? And if so, which of these is preferred by the project team?

It seems to me that 1 is cleaner for developer happiness, but 2 may have a slightly higher user happiness. And if turbo is added, the problem with either solution is temporary. I am happy to go with whatever solution y'all prefer.

Reply:
Maybe I am over-complicating this! Another solution I am thinking of is:
- Revert to https://github.com/rubyforgood/casa/commit/0f61c0ed4a74c058c58b31719cc6cef278c2cea3
- No Stimulus controller, just bootstrap classes
- When checking for filterrific params to determine collapse state, ignore the "surfaced filter" params. I could just reject them from the hash.

That way, we don't expand the menu when just checking "Hide draft". The only downside is that if you were for some reason expanding the menu and only submitted hide draft, it would collapse, but I think that is a niche situation.